lua/tardis/sh_interiors.lua

```diff
@@ -367,6 +560,145 @@ local function AddVersionSubMenu(dmenu, version)
+    end)
+    setting_button:SetIsCheckable(true)
+
+    function setting_button:Think()
+        local value = TARDIS:GetIntCustomSetting(int_id, setting_id, ply, false)
+        if self:GetChecked() ~= value then
+            self:SetChecked(value)
+        end
+    end
+
+    return setting_button
+end
+
+local function AddMenuListSetting(dmenu, int_id, setting_id, name, options)
+    local ply = LocalPlayer()
+
+    local submenu = dmenu:AddSubMenu(name, nil)
+
+    local option_buttons = {}
+
+    if not options then return end
+    for option_value, option_text in SortedPairsByValue(options) do
+
+        local option_button = submenu:AddOption(option_text, function(self)
+            TARDIS:SetIntCustomSetting(int_id, setting_id, option_value, ply)
+        end)
+        option_button:SetIsCheckable(true)
+
+        table.insert(option_buttons, {option_value, option_button})
+    end
+
+    function submenu:Think()
+        local value = TARDIS:GetIntCustomSetting(int_id, setting_id, ply)
+        for i,v in ipairs(option_buttons) do
+            v[2]:SetChecked(value == v[1])
+        end
+    end
+
+end
+
+local function AddSettingsSubmenu(parent, int_id)
+    local int_id = TARDIS:GetMainVersionId(int_id)
+
+    local metadata = TARDIS:GetInterior(int_id)
+    if not metadata then return end
+    local versions = metadata.Versions
+    local custom_settings = metadata.CustomSettings
+    local ply = LocalPlayer()
+
+    local other_versions_exist = not table.IsEmpty(versions.other)
+    local custom_versions_exist = not table.IsEmpty(versions.custom)
+    local main_version_double = (versions.main.classic_doors_id ~= nil)
+
+    local versions_exist = other_versions_exist or custom_versions_exist or main_version_double
+    local dmenu = parent:AddSubMenu("Settings", nil)
+
+    if versions_exist then
+
+        local option_versions = {
+            ["main"] = "  Default", -- spaces are intentional (sorting)
+        }
+
+        if not table.IsEmpty(versions.custom) then
+            option_versions["random_custom"] = "  Random (original & custom versions)"
+            option_versions["random"] = "  Random (original versions only)"
+        elseif not table.IsEmpty(versions.other) then
+            option_versions["random"] = "  Random"
+        end
+
+        local function add_version_option(option_id, option_name)
+            if not option_versions[option_id] then
+                option_versions[option_id] = option_name
+            end
+        end
+
+        if versions.main.classic_doors_id then
+            add_version_option(versions.main.classic_doors_id, "  Classic doors version") -- spaces are intentional (sorting)
+            add_version_option(versions.main.double_doors_id, "  Double doors version")
+        end
+
+        for k,v in SortedPairs(versions.other) do
+            if v.classic_doors_id then
+                add_version_option(v.classic_doors_id, "  " .. v.name .. " (classic doors)")
```

Comment:
Why does these other / custom versions have spaces in front of them as well? If everything has spaces doesn't that kind of defeat the point?

Reply:
I warned you about this in DMs
They are not all spaces

Some are spaces, some are `alt+255` symbols (aka "unbreakable gaps")
Some are a combination of a space and this symbol (thus, two symbols in each instance)

We keep the options sorted, but also aligned vertically

Reply:
Oh okay I see